File: swig/ruby/tests/arch.rb
#
# Arch
#

$:.unshift File.expand_path(File.join(File.dirname(__FILE__),"..","..","..","build","swig","ruby"))

require 'test/unit'
require 'zypp'

class Zypp::Arch
  include Comparable
end

class ArchTest < Test::Unit::TestCase
  def test_arch
    # define i386, a builtin
    
    a = Zypp::Arch.new("i386")
    assert a
    assert_equal "i386", a.to_s
    assert_equal true, a.is_builtin
    
    # i486 is 'bigger' than i386
    
    b = Zypp::Arch.new("i486")
    assert b
    assert_equal "i486", b.to_s
    assert b.is_builtin
    if Zypp::VERSION > 800
      assert_equal a, b.base_arch
    end
    assert a < b
    assert a.compatible_with?(b)

    # A new, adventurous architecture
    z = Zypp::Arch.new("xyzzy")
    assert z
    assert_equal "xyzzy", z.to_s
    assert_equal false, z.is_builtin
    
    # predefined archs
    assert_equal Zypp::Arch.new("noarch"), Zypp::Arch.noarch 
    assert_equal a, Zypp::Arch.i386
    assert_equal b, Zypp::Arch.i486
    assert_equal Zypp::Arch.new("i586"), Zypp::Arch.i586
    assert_equal Zypp::Arch.new("i686"), Zypp::Arch.i686
    assert_equal Zypp::Arch.new("x86_64"), Zypp::Arch.x86_64
    assert_equal Zypp::Arch.new("ia64"), Zypp::Arch.ia64
    assert_equal Zypp::Arch.new("ppc"), Zypp::Arch.ppc
    assert_equal Zypp::Arch.new("ppc64"), Zypp::Arch.ppc64
    assert_equal Zypp::Arch.new("s390"), Zypp::Arch.s390
    assert_equal Zypp::Arch.new("s390x"), Zypp::Arch.s390x
  end
end

File: swig/python/tests/arch.py
#
# Arch
#
import unittest

import os 
cwd = os.path.abspath(os.path.dirname(__file__)) 

import sys
sys.path.insert(0, cwd + "/../../../build/swig/python")
from zypp import Arch

class TestSequenceFunctions(unittest.TestCase):
    
  def testarch(self):
    a = Arch("i386")
    assert a
    assert "i386" == a.__str__()
    assert a.is_builtin()
    
    b = Arch("i486")
    assert b
    assert "i486" == b.__str__()
    assert b.is_builtin()
    assert a == b.base_arch()
    assert a < b
    assert a.compatible_with(b)

    z = Arch("xyzzy")
    assert z
    assert "xyzzy" == z.__str__()
    assert not z.is_builtin()

    assert Arch("noarch") == Arch.noarch()
    assert a, Arch.i386()
    assert b, Arch.i486()
    assert Arch("i586") == Arch.i586()
    assert Arch("i686") == Arch.i686()
    assert Arch("x86_64") == Arch.x86_64()
    assert Arch("ia64") == Arch.ia64()
    assert Arch("ppc") == Arch.ppc()
    assert Arch("ppc64") == Arch.ppc64()
    assert Arch("s390") == Arch.s390()
    assert Arch("s390x") == Arch.s390x()
if __name__ == '__main__':
  unittest.main()
